### ISSUE TITLE
Adding new field to class during project

### ISSUE THREAD

**Issue by adipso-jerome** [Oyvindkg/swiftydb#9]
Hello,
When adding a new field to my class, it seams to not be create into database. If i try to make a print of that new field, it display error "fatal error: unexpectedly found nil while unwrapping an Optional value"

**Comment by Oyvindkg** [comment]
At the moment, migration isn't automated. I have been a little busy the last few months. 

I have a few days off now, so I can probably fix it before Saturday, if that would suffice?

**Comment by adipso-jerome** [comment]
It's perfect thanks :)

**Comment by Oyvindkg** [comment]
I am spending my Easter deep in the Norwegian mountains, and I seem to have lost my internet connection (writing this on my phone). I'm afraid it will be a little delayed. 

As SQLite's ALTER TABLE only supports adding columns and renaming tables, I'm not sure how I want to implement it at the moment. 

If you need it asap, or you want to contribute, you are welcome to give it a try and send a pull request! :)

**Comment by adipso-jerome** [comment]
ok for the delay, i will wait a little.
Here I find Alter table in sqlite : http://stackoverflow.com/questions/4253804/insert-new-column-into-table-in-sqlite
Sorry, I can't help you, I have already my own project to go ahead :)

**Comment by Oyvindkg** [comment]
If automatic migration is added, I think it should at least support adding and removing columns. This isn't possible using SQLites alter table. 

I understand. Best of luck with your project :)

**Comment by adipso-jerome** [comment]
Its true, removing is not support by sqlite. Since its a sqlite problem, people will understand that the field they just delete from class won't be delete from table (even if they don't use swiftydb, the problem will be the same).

**Comment by groue** [comment]
FYI: you can perform any alteration you want to an SQLite table, including removing columns: https://www.sqlite.org/lang_altertable.html#otheralter

**Comment by Oyvindkg** [comment]
Yes, that would be the naïve, and probably the best approach. As I didn't have a chance to work on this during the Easter due to connection problems, and I am quite busy until May. I'll push a new version as soon as possible

**Comment by Oyvindkg** [comment]
The implementation is mostly done, but it needs to be tested

**Comment by izouxv** [comment]
https://www.techonthenet.com/sqlite/tables/alter_table.php

this will help you

**Comment by Oyvindkg** [comment]
Thank you @izouxv. 

[2.0](https://github.com/Oyvindkg/swiftydb/tree/dev) will apply the described process to automatically reflect updated types.
